pkg/anypoint/client.go

```diff
@@ -59,33 +60,66 @@ type ListAssetClient interface {
 
 // AnypointClient is the client for interacting with Mulesoft Anypoint.
 type AnypointClient struct {
-	baseURL     string
-	username    string
-	password    string
-	lifetime    time.Duration
-	apiClient   coreapi.Client
-	auth        Auth
-	environment *Environment
-	orgName     string
+	apiClient       coreapi.Client
+	auth            Auth
+	baseURL         string
+	clientID        string
+	clientSecret    string
+	environment     *Environment
+	environmentName string
+	lifetime        time.Duration
+	orgName         string
+	password        string
+	username        string
 }
 
+// ClientOptions callback for overriding the default config of the client
 type ClientOptions func(*AnypointClient)
 
 // NewClient creates a new client for interacting with Mulesoft.
-func NewClient(mulesoftConfig *config.MulesoftConfig, options ...ClientOptions) *AnypointClient {
-	client := &AnypointClient{}
+func NewClient(mulesoftConfig *config.MulesoftConfig, registerHealth hc.RegisterHealth, options ...ClientOptions) (*AnypointClient, error) {
+	client := &AnypointClient{
+		apiClient:       nil,
+		auth:            nil,
+		baseURL:         mulesoftConfig.AnypointExchangeURL,
+		clientID:        mulesoftConfig.ClientID,
+		clientSecret:    mulesoftConfig.ClientSecret,
+		environmentName: mulesoftConfig.Environment,
+		environment:     nil,
+		lifetime:        mulesoftConfig.SessionLifetime,
+		orgName:         mulesoftConfig.OrgName,
+		password:        mulesoftConfig.Password,
+		username:        mulesoftConfig.Username,
+	}
+
 	// Create a new client before invoking additional options, which may want to override the client
 	client.apiClient = coreapi.NewClient(mulesoftConfig.TLS, mulesoftConfig.ProxyURL)
 
 	for _, o := range options {
 		o(client)
 	}
-	client.OnConfigChange(mulesoftConfig)
```

Comment:
Same thing here. Testing was difficult when the constructor called OnConfigChange.